### DISCUSSION TITLE
Support Type for `route.ts` parameter `context`

### DISCUSSION THREAD

**Discussion by devjiwonchoi** [vercel/next.js#53690] [Ideas]
### Goals

Support type for `route.ts` parameter `context`, assuming extensions of `context`.

> Add `NextContext` type for `route.ts`

```ts
export type NextContext = {
  params: {
    slug: string[]
    // Assuming additional features?
  }
  // Assuming additional features?
}
```

```ts
// app/blog/[...slug]/route.ts
import type { NextContext } from 'next/server'

export async function GET(request: Request, context: NextContext}) {
 const { params } = context // '/blog/1/2' returns { slug: ['1', '2'] }
}
```

### Alternative

Manually add type for `context`

```ts
// app/blog/[...slug]/route.ts

export async function GET(request: Request, context: { params: { slug: string[] } }) {
 const { params } = context // '/blog/1/2' returns { slug: ['1', '2'] }
}
```

### Proposal

I am willing to contribute.

**Comment by devjiwonchoi** [comment]
Assuming this is the current type passed to context?

https://github.com/vercel/next.js/blob/654bd8c80339646beda5405511c62702f54c08df/packages/next/src/server/future/route-modules/app-route/module.ts#L64-L70

**Comment by ptvty** [comment]
Any updates on this?
It's a pain to create types manually for large projects with lots of routes, while it can be inferred automatically form the directory structure.

AFAIK TanStack Start is offering this using a Codegen that monitors changes in the file system and generates types accordingly. (Check `routeTree.gen.ts` file)
I'm willing to contribute.
